radio/src/storage/sdcard_yaml.cpp

```diff
@@ -92,15 +138,44 @@ void storageCreateModelsList()
 // SDCARD storage interface
 //
 
+static const char * attemptLoad(const char *filename, ChecksumResult* checksum_status)
+{
+  YamlTreeWalker tree;
+  tree.reset(get_radiodata_nodes(), (uint8_t*)&g_eeGeneral);
+  return readYamlFile(filename, YamlTreeWalker::get_parser_calls(), &tree, checksum_status);
+}
+
 const char * loadRadioSettingsYaml()
 {
     // YAML reader
     TRACE("YAML radio settings reader");
 
-    YamlTreeWalker tree;
-    tree.reset(get_radiodata_nodes(), (uint8_t*)&g_eeGeneral);
-
-    return readYamlFile(RADIO_SETTINGS_YAML_PATH, YamlTreeWalker::get_parser_calls(), &tree);
+    ChecksumResult checksum_status;
+    const char* p = attemptLoad(RADIO_SETTINGS_YAML_PATH, &checksum_status);
+
+    if(checksum_status != ChecksumResult::Success) {
+      FRESULT result = FR_OK;
+      TRACE("radio settings: checksum check failed");
+      if (g_eeGeneral.manuallyEdited) {
+        TRACE("File has been manually edited - ignoring checksum mismatch");
+        g_eeGeneral.manuallyEdited = 0;
+        storageDirty(EE_GENERAL);   // Trigger a save on sucessfull recovery
+      } else {
+        TRACE("File is corrupted, attempting alternative file");
+        result = f_rename(RADIO_SETTINGS_YAML_PATH, RADIO_SETTINGS_ERRORFILE_YAML_PATH); // Save corrupted file for later analysis
+        p = attemptLoad(RADIO_SETTINGS_TMPFILE_YAML_PATH, &checksum_status);
+        if (p == NULL) {
```

Comment:
somehow I always get an unrecoverable error, when I only have a radio_new.yaml, but no radio.yml

Reply:
I assume this is due to the missing check in line 154 for the reading to fail.
I am working on a fix right now :-)

Reply:
Looking at the code I realized a missing check for the temp file's checksum to be correct. I am adding that to my update and push it. I will try to flash a radio tomorrow.